lib/shoryuken/environment_loader.rb

```diff
@@ -66,8 +66,10 @@ def load_rails
       else
         # Painful contortions, see 1791 for discussion
         require File.expand_path('config/application.rb')
-        ::Rails::Application.initializer 'shoryuken.eager_load' do
-          ::Rails.application.config.eager_load = true
+        if ::Rails::VERSION::MAJOR == 4
```

Comment:
@tap87 for keeping compatibility, I'm wondering if instead of checking for Rails 4, we should check for Rails greater than or equals to 5, so all the previous version, would continue to work in the same they are now, given the issue seems to be only with Rails 5 and greater. 

BTW I'm using shoryuken with Rails 5 and it's working fine.

Reply:
Yes, we could take that approach. I tried to follow sidekiq here since that is were the existing code was adapted from. https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/cli.rb#L261